File: zemiao/80.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = 0
        i, n, temp = 0, len(nums), 0
        while i < n:
            if i > 0 and nums[i] == nums[i-1]:
                temp += 1
            else:
                temp = 0
            
            if temp < 2:
                nums[res] = nums[i]
                res += 1
            i += 1
                
        return res
                
        

File: tnzhou/80.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 0)
            return 0;
        int curr = nums[0];
        int cnt = 1;
        int leftptr = 1;
        for (int i = 1; i < nums.size(); i ++) {
            if (nums[i] == curr) {
                if (cnt >= 2) {
                    continue;
                } else {
                    cnt ++;
                    nums[leftptr++] = nums[i];
                }
            } else {
                curr = nums[i];
                cnt = 1;
                nums[leftptr++] = nums[i];
            }
        }
        nums.resize(leftptr);
        return nums.size();
    }
};
